internal/pkg/util/fs/helper.go

```diff
@@ -14,9 +14,36 @@ import (
 	"strings"
 	"syscall"
 
+	"github.com/sylabs/singularity/internal/pkg/sylog"
 	"golang.org/x/sys/unix"
 )
 
+// EnsureFileWithPermission takes a file path, and 1. Creates it with
+// the specified permission, or 2. ensures a file is the specified
+// permission.
+func EnsureFileWithPermission(fn string, mode os.FileMode) error {
+	fs, err := os.OpenFile(fn, os.O_CREATE, mode)
```

Comment:
This is not guaranteed to produce a file with the requested mode, dependent on umask. It could create a file with more restrictive permissions than requested, once the umask is applied.

Note in the old removed code the umask is being set explicitly. To guarantee mode you would need an open umask, or to do a separate `chmod` step here.

Reply:
I dont want to add the umask change here, because depending on the specified file mode, the umask will require a different value, (afaik) so a `Chmod` would be better in this case, imo.

Right after creating/opening the file, we check the file permission, if its incorrect, then we use chmod. Line ~39.

Reply:
Yeah - understood. It's a little unusual to me that we would then see a message about a permission being wrong for a file we just created, but I can live with that here.

Reply:
Makes sense, but also, that unusual warning should only appear with a unusual umask :wink: 

Would you like it better if we always run `Chmod` right after opening/creating the file? _that may not be the best solution as most of the time we will not need to run Chmod..._